tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/type-test-expr.bal

```diff
@@ -1227,6 +1227,28 @@ function testBuiltInSubTypeTypeTestAgainstFiniteType() {
     assertFalse(b is Ints);
 }
 
+public type SomeRecord record {
+    int intField;
+};
+
+public function testIfElseWithTypeTest() {
```

Comment:
Shall we add a few more tests? For example, scenarios where multiple variables are reset, different variables are reset in different blocks, different nesting combinations (even though they are effectively an if block and else block with else-ifs also in the else), etc,? I think we should add some negative tests too (e.g., trying to use it as the narrowed type after resetting).

Reply:
will add